roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -77,14 +78,24 @@ impl ParseClientJobDeclarationMessages for JobDeclaratorDownstream {
                 .unwrap()
                 .unwrap();
             let mut txs_in_job = vec![];
+            let mut txs_to_retrieve: Vec<(String, usize)> = vec![];
             let mut missing_txs = vec![];
 
             for (i, sid) in short_hash_list.iter().enumerate() {
                 let sid_: [u8; 6] = sid.to_vec().try_into().unwrap();
-                if let Some(tx_data) = short_id_mempool.get(&sid_) {
-                    txs_in_job.push(tx_data.clone());
-                } else {
-                    missing_txs.push(i as u16);
+                match short_id_mempool.get(&sid_) {
+                    Some(tx_data) => match &tx_data.tx {
+                        Some(tx) => {
+                            if i >= txs_in_job.len() {
```

Comment:
I need to do it because we need to maintain the same order defined in `DeclaredMiningJob` message in the `txs_in_job`. Using `push()` method, if a tx is not present in current JDS mempool (and so it's added to the `missing_txs` vec), another one would be inserted into that specific index (causing a wrong tx order in `declared_mining_job`)

Reply:
I don't think that it is as you are describing. Here they will be repositioned correctly. If you have (0, tx0), (1, tx1), (2, tx2), (3, _), (4, tx4), (5, _) and (6, tx6) then it will be
`
tx_in_job = vec![tx0, tx1, tx2, tx4, tx6];
missing_ts = vec![3, 5]
`
then when you receive the message PMTS
tx3 is inserted in tx_in_job[3], which becomes
```
assert_eq[tx_in_job, vec![tx0, tx1, tx3, tx2, tx4, tx6]]
```
and then tx5 is inserted in tx_in_job[5]
```
assert_eq[tx_in_job, vec![tx0, tx1, tx3, tx2, tx4, tx5, tx6]]
```
It is tricky and we should find a way to manage it better, possibly to store each transaction with its index in the block payload. Like it is now works by it is very error prone